docs/proposals/20210614-podunavailablebudget.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Pod Unavailable Budget
+authors:
+  - "@zmberg"
+reviewers:
+  - "@Fei-Guo"
+  - "@furykerry"
+  - "@FillZpp"
+creation-date: 2021-06-14
+last-updated: 2021-06-16
+status: implementable
+---
+
+# Pod Unavailable Budget Feature
+
+- In voluntary disruption scenarios, PodUnavailableBudget can achieve the effect of preventing application disruption or SLA degradation, which greatly improves the high availability of application services.
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Motivation](#motivation)
+  - [Proposal](#proposal)
+    - [API Definition](#api-definition)
+    - [Implement](#implement)
+    - [Comparison with PDB](#comparison-with-pdb)
+  - [User Cases](#user-cases)
+    - [A Sample Case](#a-sample-case)
+  - [Implementation History](#implementation-history)
+
+## Motivation
+In kubernetes, powerful workloads (e.g., deployment, statefulset, etc.) are often used to define and manage applications.
+It not only manages the number of pods, but also provides powerful rolling capabilities such as MaxUnavailable, Partition, etc.
+However, workload can only help the application reach the desired number of pods as fast as possible, but does not guarantee that the number of available pods will remain at the desired value at any given moment.
+
+Kubernetes offers [Pod Disruption Budget](https://kubernetes.io/docs/tasks/run-application/configure-pdb/) to help you run highly available applications even when you introduce frequent [voluntary disruptions](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/).
+PDB limits the number of Pods of a replicated application that are down simultaneously from voluntary disruptions. However, it can only constrain the voluntary disruption triggered by the [Eviction API](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/#eviction-api).
+For example, when you run kubectl drain, the tool tries to evict all of the Pods on the Node you're taking out of service.
+
+In the following voluntary disruption scenarios, there are still business disruption or SLA degradation situations:
+1. The application owner update deployment's pod template for general upgrading, while cluster administrator drain nodes to scale the cluster down(learn about [Cluster Autoscaling](https://github.com/kubernetes/autoscaler/#readme)).
+2. The middleware team is using [SidecarSet](https://openkruise.io/en-us/docs/sidecarset.html) to rolling upgrade the sidecar containers of the cluster, e.g. ServiceMesh envoy, while HPA triggers the scale-down of business applications.
```

Comment:
plz add the case for pod eviction initiated by admin, pod eviction is a more general case which occurs even if no kruise workload is used.